armory/metrics/poisoning.py

```diff
@@ -0,0 +1,263 @@
+    },
+    "resisc10_silhouette_model": {
+        "model_kwargs": {
+            "data_means": [0.39382024, 0.4159701, 0.40887499],
+            "data_stds": [0.18931773, 0.18901625, 0.19651154],
+            "num_classes": 10,
+        },
+        "module": "armory.baseline_models.pytorch.resnet18_bean_regularization",
+        "name": "get_model",
+        "weights_file": "resisc10_explanatory_model_resnet18_bean.pt",
+    },
+}
+
+DEVICE = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+
+
+class ExplanatoryModel:
+    def __init__(
+        self,
+        explanatory_model,
+        resize_image=True,
+        size=(224, 224),
+        resample=Image.BILINEAR,
+        device=DEVICE,
+    ):
+        if not callable(explanatory_model):
+            raise ValueError(f"explanatory_model {explanatory_model} is not callable")
+        self.explanatory_model = explanatory_model
+        self.resize_image = bool(resize_image)
+        self.size = size
+        self.resample = resample
+        self.device = device
+
+    @classmethod
+    def from_config(cls, model_config, **kwargs):
+        if isinstance(model_config, str):
+            if model_config not in EXPLANATORY_MODEL_CONFIGS:
+                raise ValueError(
+                    f"model_config {model_config}, if a str, must be in {EXPLANATORY_MODEL_CONFIGS.keys()}"
+                )
+            model_config = EXPLANATORY_MODEL_CONFIGS[model_config]
+        if not isinstance(model_config, dict):
+            raise ValueError(
+                f"model_config {model_config} must be a str or dict, not {type(model_config)}"
+            )
+        model_config = copy.copy(model_config)
+        model_config.update(kwargs)  # override config with kwargs
+        keys = ("module", "name", "weights_file")
+        for k in keys:
+            if k not in model_config:
+                raise ValueError(f"config key {k} is required")
+        module, name, weights_file = (model_config.pop(k) for k in keys)
+        model_kwargs = model_config.pop("model_kwargs", {})
+
+        weights_path = maybe_download_weights_from_s3(
+            weights_file, auto_expand_tars=True
+        )
+        model_module = import_module(module)
+        model_fn = getattr(model_module, name)
+        explanatory_model = model_fn(weights_path, **model_kwargs)
+
+        return cls(explanatory_model, **model_config)
+
+    def get_activations(self, x):
+        """
+        Return array of activations from input batch x
+        """
+        activations = []
+        with torch.no_grad():
+            x = self.preprocess(x)
+            activation, _ = self.explanatory_model(x)
+            activations.append(activation.detach().cpu().numpy())
+        return np.concatenate(activations)
+
+    @staticmethod
+    def _preprocess(
+        x, resize_image=True, size=(224, 224), resample=Image.BILINEAR, device=DEVICE
+    ):
+        if isinstance(x.dtype, np.floating):
```

Comment:
Not sure exactly what the input is expected to be at this point, but this returns false when x.dtype == np.float32 or np.float64.

It appears that `x.dtype == np.floating` returns true if `x.dtype == np.float64` but not if `x.dtype == np.float32`.

I believe armory data is loaded as np.float32.

Reply:
The input should either be `np.uint8` or any numpy floating type `np.float32`, `np.float64`, etc.